spock-specs/src/test/groovy/org/spockframework/mock/runtime/MockMakerRegistrySpec.groovy

```diff
@@ -0,0 +1,383 @@
+      getId() >> "sameprio-mock-maker"
+      getPriority() >> 100
+      getCapabilities() >> [MockMakerCapability.INTERFACE]
+    }
+    def byteBuddy = new ByteBuddyMockMaker()
+    def sampleMockMaker = Mock(IMockMaker) {
+      getId() >> "sample-mock-maker"
+      getPriority() >> 150
+      getCapabilities() >> [MockMakerCapability.INTERFACE]
+    }
+    def cglib = new CglibMockMaker()
+    when:
+    def r = new MockMakerRegistry([mockSamePriorityProxy, byteBuddy, cglib, sampleMockMaker, proxy], "sample-mock-maker")
+    then:
+    r.getMakerList() == [proxy, sampleMockMaker, byteBuddy, mockSamePriorityProxy, cglib]
+  }
+
+  def "MockMaker sorting with preferred cglib"() {
+    given:
+    def proxy = new JavaProxyMockMaker()
+    def byteBuddy = new ByteBuddyMockMaker()
+    def cglib = new CglibMockMaker()
+    when:
+    def r = new MockMakerRegistry([byteBuddy, cglib, proxy], MockMakers.cglib.mockMakerId)
+    then:
+    r.getMakerList() == [proxy, cglib, byteBuddy]
+  }
+
+  def "MockMaker non existing preferred ID"() {
+    when:
+    new MockMakerRegistry([new JavaProxyMockMaker()], MockMakers.cglib.mockMakerId)
+    then:
+    def ex = thrown(IllegalStateException)
+    ex.message == "No IMockMaker with ID 'cglib-mock-maker' exists, but was request via mockMaker.preferredMockMaker configuration. Is a runtime dependency missing?"
+  }
+
+  def "MockMaker loaded from ServiceLoader"() {
+    when:
+    def makers = defaultMockMakerRegistry.getMakerList()
+    then:
+    makers[0] instanceof JavaProxyMockMaker
+    makers[1] instanceof ByteBuddyMockMaker
+    makers[2] instanceof CglibMockMaker
+  }
+
+  def "Mock with unknown MockMaker"() {
+    when:
+    Mock(mockMaker: IMockMakerSettings.simple("unknown-mock-maker"), Runnable)
+    then:
+    def ex = thrown(CannotCreateMockException)
+    ex.message == "Cannot create mock for interface java.lang.Runnable because MockMaker with ID 'unknown-mock-maker' does not exist."
+  }
+
+  def "asMockOrNull() asks all IMockMakers"() {
+    given:
+    def maker = Mock(IMockMaker) {
+      getId() >> MockMakers.byteBuddy.mockMakerId
+      getPriority() >> 2
+      getCapabilities() >> [MockMakerCapability.INTERFACE]
+    }
+    IMockObject mockObj = Mock(IMockObject)
+    def registry = new MockMakerRegistry([maker], null)
+    def obj = "mockObj"
+    when: "The registry will ask the maker, because the obj does not implement ISpockMockObject"
+    registry.asMockOrNull(obj)
+    then:
+    1 * maker.asMockOrNull(obj) >> mockObj
+  }
+
+  def "Mock final class shall currently fail"() {
+    when:
+    Mock(StringBuilder)
+    then:
+    def ex = thrown(CannotCreateMockException)
+    ex.message == """Cannot create mock for class java.lang.StringBuilder.
+java-proxy-mock-maker: Cannot mock classes.
+byte-buddy-mock-maker: Cannot mock final classes.
+cglib-mock-maker: Cannot mock final classes."""
+  }
+
+  def "ITypeMockability.MOCKABLE"() {
```

Comment:
What is `ITypeMockability`?
And what does the test test? The name is not too chatty.

Reply:
That was the old name of the `IMockabilityResult`, I forgot to rename the test case.